partiql-lang/src/main/kotlin/org/partiql/lang/graph/SimpleGraph.kt

```diff
@@ -5,27 +5,31 @@ import org.partiql.lang.eval.ExprValue
 internal typealias Labels = Set<String>
 
 /** A straightforward implementation of in-memory graphs. */
-internal class SimpleGraph(
-    internal val nodes: List<Node>,
-    internal val directed: List<Triple<Node, EdgeDirected, Node>>,
-    internal val undir: List<Triple<Node, EdgeUndir, Node>> // order of Nodes in the Triple "doesn't matter"
+class SimpleGraph(
+    val nodes: List<Node>,
+    val directed: List<Triple<Node, EdgeDirected, Node>>,
+    val undir: List<Triple<Node, EdgeUndir, Node>> // order of Nodes in the Triple "doesn't matter"
 ) : Graph {
```

Comment:
So, `SimpleGraph` made internal
       ==> `ExternalGraphReader` made internal (because `read()` returns `SimpleGraph`)
          ==> problem in `Shell` -- can't call `ExternalGraphReader.read(graphIonText)`
Now, the latter is within
```
   ExprValue.newGraph(ExternalGraphReader.read(graphIonText))
```
So, what can be done is adding to `ExprValue` companion:
```
        @JvmStatic
        fun newGraphFromTextIon(graphIonText: String): ExprValue =
            ExprValue.newGraph(ExternalGraphReader.read(graphIonText))
```
and calling this new method in `Shell` instead.  Then the switches to internal work out.

In my view, this is trading one unpleasantness (internal/public for `SimpleGraph`, etc) for another (pulling a dependency on `ExternalGraphReader` probably too deep into the data model).

Do you see a better resolution? Maybe there is a better location for this new method?

Reply:
To me, since `ExternalGraphReader` is only used within the CLI and tests, it makes the most sense to me to move it to the CLI. Same thing with `SimpleGraph`. And, you can still use `ExprValue.newGraph()`.

Similarly, `SimpleGraph` is only used by `ExternalGraphReader`, tests, and `BaseExprValue` (as a companion object to create an empty graph). You could move `SimpleGraph` to the CLI project, and `BaseExprValue` can just have the companion object be:
```kotlin
companion object {
    val empty: Graph = object : Graph
}
```

Especially since Graph doesn't have any public methods to override.

And, then, when you want to create a `GraphExprValue` in the CLI, all you need to write is:

```kotlin
// IN CLI PROJECT
val graph: SimpleGraph = ExternalGraphReader.read(someFile) // Both classes are internal to CLI
val graphExprValue = ExprValue.newGraph(graph) // ExprValue.newGraph comes from the core library
```

---

My main point is that `ExternalGraphReader` and `SimpleGraph` are implementation-specific, and they don't need to be made public in the core library. I don't see anywhere in this PR that we need to expose their implementations.

Reply:
So, you suggest to move `SimpleGraph` and `ExternalGraphReader` from partiql-lang to `partiql-cli`, so that unit tests of partiql-lang become dependent on partiql-cli?  
I did not find out whether this can even work in our setup, but it just does not look right... 

Some of your reasoning (assumptions of things being used in limited number of places or classes not having methods) might also depend on the accident of this PR having been extracted from a larger body of work (see downstream #1074). 

I am with you on the need to find proper homes for the parts of this code, but do not yet see clearly where they should be.  I believe this would be best done later.  From the points brought up so far, it might be that `SimpleGraph` and `ExternalGraphReader` might even have to live in their own library?

If this is really important to resolve right now, I request moving the discussion, argumentation, and experimentation into PR #1074, which offers a more compete context of what is at play.  If you do so, I am ok keeping this #1073 open until the matter is resolved in #1074.